pkg/deviceplugin/lister.go

```diff
@@ -66,17 +68,86 @@ func (ml macvtapLister) Discover(pluginListCh chan dpm.PluginNameList) {
 		plugins = append(plugins, macvtapConfig.Name)
 	}
 
-	pluginListCh <- plugins
+	if len(plugins) > 0 {
+		pluginListCh <- plugins
+	}
+	return config, nil
+}
+
+func discoverByLinks(pluginListCh chan dpm.PluginNameList, nsPath string) {
+	// To know when the manager is stoping, we need to read from pluginListCh.
+	// We avoid reading our own updates by using a middle channel.
+	parentListCh := make(chan []string)
+	defer close(parentListCh)
+
+	sendSuitableParents := func() {
+		var linkNames []string
+		err := ns.WithNetNSPath(nsPath, func(_ ns.NetNS) error {
+			var err error
+			linkNames, err = util.FindSuitableMacvtapParents()
```

Comment:
Would it be better if we handled the WithNetNSPath inside the util function? So it is as close to the code that needs it as possible and only around that code?

Reply:
I see it the other way. One of the main issues is correct namespace handling and I feel it should not be hidden away inside utility methods. This way reads naturally, like, `find suitable macvtap parents inside this namespace`. We are not bundling a bunch of unrelated code within the block.